### DISCUSSION TITLE
50. IPv4와 IPv6를 주소, 보안, 성능 측면에서 비교해주세요.

### DISCUSSION THREAD

**Discussion by linirini** [jewelry-box/cs-interview#52] [General]
`키워드`: IPv4, IPv6, NAT, IPSec, IP 단편화

**Comment by cutehumanS2** [comment]
1. **주소 공간**
    - IPv4는 32비트 주소 체계를 사용하며, 약 43억 개의 주소만 할당 가능합니다. 주소 부족 문제를 해결하기 위해 NAT(네트워크 주소 변환)를 사용합니다.
    - IPv6는 128비트 주소 체계를 사용하여, 사실상 무제한에 가까운 주소를 할당할 수 있습니다. NAT 없이도 충분한 주소를 제공합니다.
2. **주소 표현 방식**
    - IPv4는 점으로 구분된 십진수 형식을 사용합니다.
    - IPv6은 콜론으로 구분된 16진수 형식을 사용합니다.
3. **보안**
    - IPv4는 IPSec을 선택적으로 구현하며, 기본적으로 암호화를 지원하지 않습니다.
    - IPv6는 IPSec을 기본적으로 지원하여, 패킷 인증과 암호화를 통해 보안을 강화합니다.
4. **성능**
    - IPv4는 라우터에서도 단편화를 수행하기 때문에 라우터의 처리 부담이 증가할 수 있습니다.
    - IPv6는 단편화가 송신 측에서만 수행되기 때문에 라우터의 부담이 줄어듭니다. 또한, 헤더 구조가 간소화되어 라우팅 효율이 향상됩니다.

<br><br>
[참고한 자료]
[**정보통신기술용어해설 - IP Fragmentation, IP Segmentation   IP 단편화, IP 조각화, IPv4 단편화, IPv6 단편화**](http://www.ktword.co.kr/test/view/view.php?m_temp1=5236&id=1003)

그리고 gpt ㅎ ㅎ..

[추가 학습 자료]
- IPSec:  https://www.cloudflare.com/ko-kr/learning/network-layer/what-is-ipsec/
- NAT: [[https://inpa.tistory.com/entry/WEB-🌐-NAT-란-무엇인가](https://inpa.tistory.com/entry/WEB-%F0%9F%8C%90-NAT-%EB%9E%80-%EB%AC%B4%EC%97%87%EC%9D%B8%EA%B0%80)](https://inpa.tistory.com/entry/WEB-%F0%9F%8C%90-NAT-%EB%9E%80-%EB%AC%B4%EC%97%87%EC%9D%B8%EA%B0%80)
